include/avif/avif.h

```diff
@@ -199,8 +199,8 @@ typedef struct avifRWData
 // clang-format on
 
 // The avifRWData input must be zero-initialized before being manipulated with these functions.
-AVIF_API void avifRWDataRealloc(avifRWData * raw, size_t newSize);
-AVIF_API void avifRWDataSet(avifRWData * raw, const uint8_t * data, size_t len);
+AVIF_API avifResult avifRWDataRealloc(avifRWData * raw, size_t newSize);
+AVIF_API avifResult avifRWDataSet(avifRWData * raw, const uint8_t * data, size_t len);
```

Comment:
I just saw one of your suggested follow-ups is "Propagate `AVIF_RESULT_OUT_OF_MEMORY` to all sites calling a function listed above (not done in this change to keep it at a reasonable size)".

Is this why there are still some calls to these functions whose return values are not checked?

Reply:
> Temporarily add __attribute__ ((__warn_unused_result__)) to the end of these function declarations. You will find some calls whose return values are not checked.

The attribute did not trigger any Clang warning on `avifRWDataRealloc()` and `avifRWDataSet()`. I assume you meant `avifImageSetMetadataExif()` etc.

> Is this why there are still some calls to these functions whose return values are not checked?

I did not want to make a lengthy change review but since there are already multiple comments, I carried on with `avifImageSet*()` and `avifCodecEncodeOutputAddSample()`.
It does not matter much as long as `avifAlloc()` aborts anyway.

Reply:
> return values are not checked

That was intended but done.

Reply:
I see. We could add a `(void)` cast to indicate the return value was ignored intentionally, but GCC 12.2.0 still warns that the return value is ignored. So I ended up asserting the return value is `AVIF_RESULT_OK`.